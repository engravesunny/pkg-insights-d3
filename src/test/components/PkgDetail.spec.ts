import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils' 
import PkgDetailVue from '@/components/PkgDetail.vue';
import 'element-plus/theme-chalk/base.css';

// PkgDetail组件测试
describe('PkgDetail test', () => {
    const data = {
        entryPackageName: '包名',
        entryVersion: '版本',
        isCircle: true,
        isMulPackage: false,
        mulPackageList: ['链接1', '链接2', '链接3'],
    };
    // 渲染测试
    test('组件渲染测试', () => {
        const wrapper = mount(PkgDetailVue, {
            props: {
                data
            }
        });
        
        expect(wrapper.html()).toMatchSnapshot();
    })
    // 属性渲染测试
    test('props属性是否正确渲染', () => {
        const wrapper = mount(PkgDetailVue, {
            props: {
                data,
            },
        });
        expect(wrapper.find('.el-descriptions__label').text()).toBe('包名');
        expect(wrapper.find('.el-descriptions__content').text()).toBe(data.entryPackageName);

        expect(wrapper.findAll('.el-descriptions__label')[1].text()).toBe('版本');
        expect(wrapper.findAll('.el-descriptions__content')[1].text()).toBe(data.entryVersion);

        expect(wrapper.findAll('.el-descriptions__label')[2].text()).toBe('是否循环依赖');
        expect(wrapper.findAll('.el-tag')[0].text()).toBe(data.isCircle ? '是' : '否');

        expect(wrapper.findAll('.el-descriptions__label')[3].text()).toBe('是否多版本');
        expect(wrapper.findAll('.el-tag')[1].text()).toBe(data.isMulPackage ? '是' : '否');

        const links = wrapper.findAll('.links-box .el-link');
        expect(links.length).toBe(data.mulPackageList.length);
        links.forEach((link, index) => {
        expect(link.text()).toBe(data.mulPackageList[index]);
        });
    })
    // 事件触发测试
    test('多版本查询 点击链接时应该触发 refresh 事件', async () => {
        const wrapper = mount(PkgDetailVue, {
            props: {
                data
            },
        });
    
        const links = wrapper.findAll('a');
        await links[0].trigger('click');
        expect(wrapper.emitted('refresh')).toBeTruthy();
        expect((wrapper.emitted('refresh') as any)[0][0]).toBe('链接1');
    
        await links[1].trigger('click');
        expect(wrapper.emitted('refresh')).toBeTruthy();
        expect((wrapper.emitted('refresh') as any)[1][0]).toBe('链接2');
    
        await links[2].trigger('click');
        expect(wrapper.emitted('refresh')).toBeTruthy();
        expect((wrapper.emitted('refresh') as any)[2][0]).toBe('链接3');
    });
    
})