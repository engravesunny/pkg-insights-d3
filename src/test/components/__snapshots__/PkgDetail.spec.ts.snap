// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PkgDetail test > 组件渲染测试 1`] = `
"<div data-v-938b9439=\\"\\" class=\\"el-descriptions\\">
  <div class=\\"el-descriptions__header\\">
    <div class=\\"el-descriptions__title\\">依赖详情</div>
    <div class=\\"el-descriptions__extra\\"></div>
  </div>
  <div class=\\"el-descriptions__body\\">
    <table class=\\"el-descriptions__table is-bordered\\">
      <tbody>
        <tr>
          <td style=\\"min-width: 150px;\\" class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">包名</td>
          <td style=\\"min-width: 150px;\\" class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"NaN\\">包名</td>
        </tr>
        <tr>
          <td class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">版本</td>
          <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"NaN\\">版本</td>
        </tr>
        <tr>
          <td class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">是否循环依赖</td>
          <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"NaN\\">
            <transition-stub data-v-938b9439=\\"\\" name=\\"el-zoom-in-center\\" appear=\\"true\\" persisted=\\"false\\" css=\\"true\\"><span class=\\"el-tag el-tag--danger el-tag--light\\"><span class=\\"el-tag__content\\">是</span>
              <!--v-if--></span>
            </transition-stub>
          </td>
        </tr>
        <tr>
          <td class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">是否多版本</td>
          <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"NaN\\">
            <transition-stub data-v-938b9439=\\"\\" name=\\"el-zoom-in-center\\" appear=\\"true\\" persisted=\\"false\\" css=\\"true\\"><span class=\\"el-tag el-tag--info el-tag--light\\"><span class=\\"el-tag__content\\">否</span>
              <!--v-if--></span>
            </transition-stub>
          </td>
        </tr>
        <tr>
          <td class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">多版本查询</td>
          <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"1\\">
            <div data-v-938b9439=\\"\\" class=\\"links-box\\"><a data-v-938b9439=\\"\\" class=\\"el-link el-link--primary\\">
                <!--v-if--><span class=\\"el-link__inner\\">链接1</span>
                <!--v-if-->
              </a><a data-v-938b9439=\\"\\" class=\\"el-link el-link--primary\\">
                <!--v-if--><span class=\\"el-link__inner\\">链接2</span>
                <!--v-if-->
              </a><a data-v-938b9439=\\"\\" class=\\"el-link el-link--primary\\">
                <!--v-if--><span class=\\"el-link__inner\\">链接3</span>
                <!--v-if-->
              </a></div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>"
`;
